crates/rome_js_analyze/src/analyzers/nursery/no_fallthrough_switch_clause.rs

```diff
@@ -47,36 +59,142 @@ declare_rule! {
+                                        has_default_clause = true;
+                                        default_clause = Some(switch_clause);
+                                    } else {
+                                        switch_clauses.push_back(switch_clause);
+                                    }
+                                }
+                                if let Some(default_clause) = default_clause.take() {
+                                    switch_clauses.push_back(default_clause);
+                                }
+                            }
+                        }
+                    }
+                    InstructionKind::Jump {
+                        conditional, block, ..
+                    } => {
+                        let jump_block_index = block.index();
+                        // Avoid cycles and redundant checks.
+                        if visited_blocks.insert(jump_block_index) {
+                            block_stack.push(jump_block_index);
+                        }
+                        // If we are in a block of a switch statements,
+                        // then any conditional jump is a case and an unconditional jump
+                        // is a default clause if the switch has a default clause.
+                        if is_switch && (conditional || has_default_clause) {
+                            // Take the unconditional jump into account only if a default clause is present.
+                            switch_clause_blocks.insert(jump_block_index);
+                            let Some(switch_clause) = switch_clauses.pop_front() else {
+                                unreachable!("Missing switch clause.")
+                            };
+                            block_to_switch_clause_range.insert(
+                                jump_block_index,
+                                if switch_clause.consequent().is_empty() {
+                                    // Ignore empty switch clauses
+                                    None
+                                } else {
+                                    Some(switch_clause.range())
+                                },
+                            );
+                        }
+                        if !conditional {
+                            // The next instructions are unreachable.
+                            break;
+                        }
+                    }
+                    InstructionKind::Return => {
+                        // The next instructions are unreachable.
+                        break;
+                    }
+                }
+            }
+            if !switch_clause_blocks.is_empty() {
+                // Analyze the found switch clauses to detect any fallthrough.
+                register_fallthrough_switch_clauses(
+                    &block_to_switch_clause_range,
+                    &visited_blocks,
```

Comment:
This can be passed without reference because they are not used anymore

Reply:
Why prefer consuming the value instead of borrowing it?

By the way, I think I could reuse `block_to_switch_clause_range` by clearing it first.

Reply:
Because you only need that value in that part of the code, and it's never used later. Doing so, you avoid calling `copied` by other functions.